Sming/Arch/Esp32/Tools/install.sh

```diff
@@ -61,22 +61,24 @@ rm -f "$IDF_PATH"
 ln -s "$IDF_CLONE_PATH" "$IDF_PATH"
 
 
-# Skip installation for CI if already present
-if [ -z "$CI_BUILD_DIR" ] || [ ! -d "$IDF_TOOLS_PATH/tools" ]; then
-
-# Install IDF tools and packages
-python3 "$IDF_PATH/tools/idf_tools.py" --non-interactive install "*elf*"
-if [ -n "$VIRTUAL_ENV" ]; then
-    unset VIRTUAL_ENV
-    unset VIRTUAL_ENV_PROMPT
-    export PATH="${PATH/$VIRTUAL_ENV\/bin:/}"
-fi
-python3 "$IDF_PATH/tools/idf_tools.py" --non-interactive install-python-env
+# Install IDF tools and packages (unless CI has restored from cache)
+if [ -n "$CI_BUILD_DIR" ] && [ -d "$IDF_TOOLS_PATH/tools" ]; then
+    printf "\n\n** Skipping IDF tools installation: '%s/tools' exists\n\n" "$IDF_TOOLS_PATH"
+else
+    # Be specific about which tools we want to install for each IDF version
+    IDF_TOOL_PACKAGES=$(tr '\n' ' ' < "$SMING_HOME/Arch/Esp32/Tools/idf_tools-${INSTALL_IDF_VER}.lst")
+    echo "Install: $IDF_TOOL_PACKAGES"
+    python3 "$IDF_PATH/tools/idf_tools.py" --non-interactive install $IDF_TOOL_PACKAGES
```

Comment:
Codacy persistently complains when variables such as `$IDF_TOOL_PACKETS` are used without quotes. I'm not clear on the best way to shut it up, but this usage is intended.

Reply:
> clear on the best way to shut it up, but this usage is intended.

I will do it. Basically you need to click on "Ignore issue" for this file.

Reply:
Done.